src/device.jl

```diff
@@ -77,23 +77,47 @@ function devices(f::Function, devlist::Union{Integer,AbstractVector})
 end
 
 function filter_free(devlist)
-    smi = readall(`nvidia-smi`)
-    if contains(smi, "No running compute processes")
-        return devlist
+    usenvml = true
+    if Libdl.find_library(["libnvidia-ml"], []) == ""
```

Comment:
I just noticed that we most likely will require to use the result of `find_library` in the ccalls below otherwise Windows won't work.